### PR TITLE
Require rng-tools to generate more entropy faster for RHEL systems.

### PR DESCRIPTION
**Pull request by sbluhm**
## What does this PR change?
**Issue:**
The Tomcat startup/rhn context can hang "forever" on creating a pg-jdbc connection (literally days).

Background:
pg-jdbc is waiting for entropy from `/dev/random` when creating the ssl connection. On RHEL there is a chance that not enough entropy is available.
This can be checked by running `cat /proc/sys/kernel/random/entropy_avail` (value < 200 is bad, above 2000 is nice).
`rngd` or `rng-tools` (random number generator daemon) increase the entropy significantly faster by tapping additional entropy sources.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: already covered

- [X] **DONE**

## Links

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by sbluhm** [comment]
This issue showed up first for me on 2022-02-11 resulting in consistent system unavailability.

**Would this fix be relevant for Leap Systems as well?**

**Comment by sbluhm** [comment]
@mcalmer , can you please flag this as a merge candidate? This one is critical for RHEL users.

**Comment by mcalmer** [comment]
Wow, i wonder if we can get similar problems with SUSE as well. 
@moio FYI

**Comment by sbluhm** [comment]
This is very system dependent. We tried on virtual machines from 1cores/2GB to 4cores/18GB.
The 1core/2GB worked only with full (default) debug logging enabled on Tomcat and rhn. Took 2 mins startup instead of 40secs. Also tried on 4cores/16GB and it worked but that machine was Oracle so might have had rng already installed. Will actually check.
Other setups were not successful.
Actually can you check if rng is installed by default on Leap?

**Comment by mcalmer** [comment]
rng is not installed on Leap. It even does not exist on SLE15.
On SLE15 we have "haveged" installed which seems to do the same, but in an old way.
But this explains, why we do not see these problems on SLE.

**Comment by mcalmer** [comment]
On SLE as well as on Leap 15, havenged is "recommeded" by the base pattern.
rng-tools exists on Leap, but not on SLE.

**Comment by sbluhm** [comment]
I have put `Recommends` here and only for RHEL8. It was needed on my setup. But might not be needed on all setups. Kernel 5.6 /dev/random does not block anymore. (https://github.com/torvalds/linux/commit/30c08efec8884fb106b8e57094baa51bb4c44e32)

What do you think about a message during setup? (i.e. rng != installed: "You do not have an enhanced random number generator installed. You might want to consider installing the package rng-tools")

File: java/spacewalk-java.spec [hunk 1]
```diff
@@ -664,6 +667,11 @@ echo "#### SYMLINKS END ####"
 %service_add_pre taskomatic.service
 %endif
 
+%post
+%if 0%{?rhel}
+systemctl start rngd
```
Comment (mcalmer) [inline java/spacewalk-java.spec:672]: While rng-tools is now only "Recommended": How does this behave if rng-tools is not installed?

Comment (sbluhm) [reply java/spacewalk-java.spec:672]: good point, forgot about that line

Comment (mcalmer) [reply java/spacewalk-java.spec]: Fine with me. :+1:

**Review by mcalmer — APPROVED** [review]

**Comment by mcalmer** [comment]
I rebased this and merged

**Pull request closed without merging**